x-pack/plugins/observability/public/hooks/use_alert_index_names.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useFetcher } from './use_fetcher';
+import { callObservabilityApi } from '../services/call_observability_api';
+
+const NO_INDEX_NAMES: string[] = [];
+
+export function useAlertIndexNames() {
+  const { data: indexNames = NO_INDEX_NAMES } = useFetcher(({ signal }) => {
+    return callObservabilityApi('GET /api/observability/rules/alerts/dynamic_index_pattern', {
+      signal,
+      params: {
+        query: {
+          namespace: 'default',
+          registrationContexts: [
+            'observability.apm',
```

Comment:
Same here for hardcoded `registrationContexts`. It seems that these values are technical filed-like?

Reply:
I understand your point here, but I would prefer to keep this inside the hook because at this point we want always the same `registrationContexts`. We can always refactor it in the future if we need to.

Reply:
@fkanout I see your point and it would be nice to have constants for the registration contexts. As part of this PR I extracted existing functionality to a new hook https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/alerts/containers/alerts_page/alerts_page.tsx#L148, so we could definitely refactor this in a follow up PR